src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -284,7 +284,9 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
 
             if (triggerBinding != null)
             {
-                functionDefinition = CreateTriggeredFunctionDefinition(triggerBinding, triggerParameterName, _executor, functionDescriptor, nonTriggerBindings, invoker, _singletonManager);
+                Type triggerValueType = triggerBinding.TriggerValueType;
+                MethodInfo createTriggeredFunctionDefinitionMethodInfo = typeof(FunctionIndexer).GetMethod("CreateTriggeredFunctionDefinitionImpl", BindingFlags.Instance | BindingFlags.NonPublic).MakeGenericMethod(triggerValueType);
```

Comment:
Yep. 

Question @mathewc @MikeStall : do we have host startup perf tests in place?

 Changing this method to instance instead of static reduces a lot of confusing clutter, but it may have been there to avoid some perf hit related to invoking a generic method (only reason I can imagine it was coded as it was)